pykechain/client.py

```diff
@@ -1105,6 +1105,106 @@ def create_activity(
+        :type part_parent_model: Part2
+        :param part_parent_instance: (O) parent Part2 object or UUID for the copied part instance(s)
+        :type part_parent_instance: Part2
+        :param part_model_rename_template: (O) renaming template for part models. Must contain "{name}"
+        :type part_model_rename_template: str
+        :param part_instance_rename_template: (O) renaming template for part instances. Must contain "{name}"
+        :type part_instance_rename_template: str
+        :param asynchronous: If true, immediately returns without activities (default = False)
+        :type asynchronous: bool
+        :return: list of cloned activities
+        :rtype: list
+        :raises APIError if cloned
+        """
+        update_name = 'activity_update_dicts'
+        activity_ids = check_list_of_base(activities, cls=Activity2, key='activities')
+        update_dicts = check_type(activity_update_dicts, dict, key=update_name) or dict()
+
+        if not all(key in activity_ids for key in update_dicts.keys()):
+            incorrect_ids = [key for key in update_dicts.keys() if key not in activity_ids]
+            raise IllegalArgumentError(
+                "The `{}` must contain updates to activities that are to be cloned. "
+                "Did not recognize the following UUIDs:\n{}".format(update_name, '\n'.join(incorrect_ids)))
+
+        elif not all(isinstance(value, dict) for value in update_dicts.values()):
+            raise IllegalArgumentError("The `{}` must be a dict of dicts.".format(update_name))
+
+        activities = [dict(id=uuid, **update_dicts.get(uuid, {})) for uuid in activity_ids]
+
+        data = dict(
+            parent_id=check_base(parent, cls=Activity2, key='parent'),
+            clone_parts=check_type(clone_parts, bool, 'clone_parts'),
+            clone_part_instances=check_type(clone_part_instances, bool, 'clone_part_instances'),
+            clone_children=check_type(clone_children, bool, 'clone_children'),
+            exclude_model_ids=check_list_of_base(excluded_parts, cls=Part2, key='excluded_models') or [],
+            part_parent_model_id=check_base(part_parent_model, Part2, 'part_parent_model'),
+            part_parent_instance_id=check_base(part_parent_instance, Part2, 'part_parent_instance'),
+            cloned_part_models_rename_template=check_type(
+                part_model_rename_template, str, 'part_model_rename_template'),
+            cloned_part_instances_rename_template=check_type(
+                part_instance_rename_template, str, 'part_instnace_rename_template'),
+            activities=activities,
+        )
+
+        if kwargs:
+            data.update(kwargs)
+
+        params = dict(API_EXTRA_PARAMS['activities'])
+        params['async_mode'] = asynchronous
+
+        response = self._request('POST', self._build_url('activities_bulk_clone'),
+                                 json=data, params=params)
```

Comment:
Just a check: the data is a full on json as the activities_fvalues is a list of dict. Ensure that the data is encoded as json and send out as application/json. 

- [ ] I did the check and I confirm that the json is properly encoded, thank you!

Reply:
This request does the parsing? I was stuck on this issue until I started using `json=data` instead of `data=data`. Is this the check you are talking about?

Reply:
yes